.github/workflows/pyright.yml

```diff
@@ -1,14 +1,28 @@
-- uses: actions/checkout@v4
-- uses: actions/setup-python@v5
-  with:
-    python-version: '3.12'
-    cache: 'pip'
+name: Pyright
+on: [pull_request]
+jobs:
+  pyright:
+    name: pyright
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: jakebailey/pyright-action@v2
+        with:
+          version: 1.1.356
+          github_token: ${{ secrets.GITHUB_TOKEN }} # You need this
```

Comment:
Where are these options documented? I haven't found it at https://github.com/jakebailey/pyright-action.

Reply:
I found it here: https://github.com/jordemort/action-pyright?tab=readme-ov-file#example-usage

But we can experiment with removing `GITHUB_TOKEN`.

Reply:
Hmm, but you're linking to a different pyright-action, right? The one you link is from `jordemort`, but the GHA workflow uses another one from `jakebailey`.

Reply:
Yeah, I combined those two. This PR is basically what we discussed a while ago about trying to make Pyright working in a branch and re-iterating. I'll fix it tomorrow.